apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -379,39 +379,31 @@ defmodule Aecore.Chain.Worker do
   defp calculate_block_acc_txs_info(block) do
     block_hash = BlockValidation.block_header_hash(block.header)
 
-    accounts =
-      for tx <- block.txs do
-        case tx.data do
-          %SpendTx{} ->
-            [tx.data.sender, tx.data.receiver]
+    accounts_unique =
+      block.txs
+      |> Enum.map(fn tx ->
+        case tx.data.type do
+          SpendTx ->
+            [tx.data.sender, tx.data.payload.receiver]
 
-          %DataTx{} ->
+          _ ->
             tx.data.sender
         end
-      end
-
-    accounts_unique = accounts |> List.flatten() |> Enum.uniq() |> List.delete(nil)
+      end)
+      |> List.flatten()
+      |> Enum.uniq()
+      |> List.delete(nil)
 
     for account <- accounts_unique, into: %{} do
-      acc_txs =
-        Enum.filter(block.txs, fn tx ->
-          case tx.data do
-            %SpendTx{} ->
-              tx.data.sender == account || tx.data.receiver == account
-
-            %DataTx{} ->
-              tx.data.sender == account
-          end
-        end)
-
-      tx_hashes =
-        Enum.map(acc_txs, fn tx ->
-          tx_bin = Serialization.pack_binary(tx)
-          :crypto.hash(:sha256, tx_bin)
-        end)
-
+      # txs associated with the given account
       tx_tuples =
-        Enum.map(tx_hashes, fn hash ->
+        block.txs
+        |> Enum.filter(fn tx ->
+          tx.data.sender == account || tx.data.payload.receiver == account
```

Comment:
don't we have the same problem here? https://github.com/aeternity/elixir-node/pull/316#discussion_r177672949

Reply:
I don't think so, because the filter will only take the transactions where the sender or the receiver is matching the account. Here we don't care if we have one or the other. Maybe it won't work if we neither have sender nor receiver, but I don't think we are going to have such scenario.